pipeline/frontend/yaml/compiler/dag_test.go

```diff
@@ -59,4 +59,74 @@ func TestConvertDAGToStages(t *testing.T) {
 	}
 	_, err = convertDAGToStages(steps, "")
 	assert.ErrorIs(t, err, &ErrStepMissingDependency{})
+
+	steps = map[string]*dagCompilerStep{
+		"echo env": {
+			position:  0,
+			name:      "echo env",
+			group:     "",
+			dependsOn: []string{""},
```

Comment:
@6543 Why do you want to support such a wrong configuration? Shouldn't we treat it like an error if someone writes: `depends_on: [""]` instead of `depends_on: []` Or is it because of this super weirdish way go-yaml is converting the slice?

Reply:
I just want to let the server follow our idea: if run and on strange edgecases make sure it still works

the linter of course could point out a warning, but if you do `depends_on: [""]` as **a naive user** i expect it to not depend on something ... since the string is empty right :laughing:

Reply:
I would just throw an error instead of supporting such configs

Reply:
Well I'm for supporting users

Reply:
Yes, but not bad habits and wrong configs. Using an empty string is just wrong IMO.

Reply:
well we now even have a good use-case, and a user intuitively try to use it that way already

Reply:
I still think it supports wrong configs. The mentioned issue was a bug solved by #3097